pkg/session/session.go

```diff
@@ -208,15 +213,56 @@ func newClient(ctx context.Context, logger logr.Logger, sessionKey string, url *
 	return c, nil
 }
 
-func clearCache(sessionKey string) {
-	sessionMU.Lock()
-	defer sessionMU.Unlock()
-	delete(sessionCache, sessionKey)
+func clearCache(logger logr.Logger, sessionKey string) {
+	if cachedSession, ok := sessionCache.Load(sessionKey); ok {
+		s := cachedSession.(*Session)
+
+		// we check for tagmanager session
+		// because calling on expired session blocks
+		session, err := s.TagManager.Session(context.Background())
+		if err != nil {
+			logger.Error(err, "unable to get tag manager session")
+		}
+		if session != nil {
+			logger.Info("found active tag manager session, logging out")
```

Comment:
The documentation for keep alive handler talks about the groutine being killed only when `Logout` is called or an error is sent. I know we are calling a `Logout` in the `clearCache()` function, but returning an error would make sure that we are not leaking goroutines in case the call to `Logout` does not go through for some reason.

Reply:
Ack